scilpy/tractograms/tests/test_streamline_operations.py

```diff
@@ -0,0 +1,352 @@
+# -*- coding: utf-8 -*-
+import os
+import tempfile
+
+import nibabel as nib
+import numpy as np
+import pytest
+
+from dipy.io.streamline import load_tractogram
+from dipy.tracking.streamlinespeed import length
+
+from scilpy.io.fetcher import fetch_data, get_testing_files_dict, get_home
+from scilpy.tractograms.streamline_operations import (
+    filter_streamlines_by_length,
+    filter_streamlines_by_total_length_per_dim,
+    resample_streamlines_num_points,
+    resample_streamlines_step_size,
+    smooth_line_gaussian,
+    smooth_line_spline)
+from scilpy.tractograms.tractogram_operations import concatenate_sft
+
+fetch_data(get_testing_files_dict(), keys=['tractograms.zip'])
+tmp_dir = tempfile.TemporaryDirectory()
+
+
+def _setup_files():
+    """ Load streamlines and masks relevant to the tests here.
+    """
+
+    os.chdir(os.path.expanduser(tmp_dir.name))
+    in_long_sft = os.path.join(get_home(), 'tractograms',
+                               'streamline_operations',
+                               'bundle_4.tck')
+    in_mid_sft = os.path.join(get_home(), 'tractograms',
+                              'streamline_operations',
+                              'bundle_4_cut_endpoints.tck')
+    in_short_sft = os.path.join(get_home(), 'tractograms',
+                                'streamline_operations',
+                                'bundle_4_cut_center.tck')
+    in_ref = os.path.join(get_home(), 'tractograms',
+                          'streamline_operations',
+                          'bundle_4_wm.nii.gz')
+
+    in_rois = os.path.join(get_home(), 'tractograms',
+                           'streamline_operations',
+                           'bundle_4_head_tail_offset.nii.gz')
+
+    # Load sft
+    long_sft = load_tractogram(in_long_sft, in_ref)
+    mid_sft = load_tractogram(in_mid_sft, in_ref)
+    short_sft = load_tractogram(in_short_sft, in_ref)
+
+    sft = concatenate_sft([long_sft, mid_sft, short_sft])
+
+    # Load mask
+    rois = nib.load(in_rois)
+    return sft, rois
+
+
+def test_filter_streamlines_by_length_max_length():
+    """ Test the filter_streamlines_by_length function with a max length.
+    """
+
+    sft, _ = _setup_files()
+
+    min_length = 0.
+    max_length = 100
+    # Filter streamlines by length and get the lengths
+    filtered_sft = filter_streamlines_by_length(
+        sft, min_length=min_length, max_length=max_length)
+    lengths = length(filtered_sft.streamlines)
+
```

Comment:
Do you know the initial lengths of your sft? Are they already all < 100?
Maybe max_length = max(of all your streamlines) - 10?

Reply:
Good thinking. Added an assertion that some streamlines were removed.